datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -213,7 +220,8 @@ impl TableProviderFactory for ListingTableFactory {
             .with_file_sort_order(cmd.order_exprs.clone())
             .with_insert_mode(insert_mode)
             .with_single_file(single_file)
-            .with_write_options(file_type_writer_options);
+            .with_write_options(file_type_writer_options)
+            .with_infinite_source(unbounded);
```

Comment:
Please remove the line with `.with_infinite_source(unbounded);` as it duplicates the logic above.

Reply:
I took the liberty of doing so in e6f1b2fe5a0219118919da5eaa24ac9293334ff3

Reply:
Thanks!